scripts/game_logic/grid.c

```diff
@@ -0,0 +1,123 @@
+/*
+ ============================================================================
+ Name        : grid_gen.c
+ Author      : Liam Tapper
+ Version     :
+ Copyright   :
+ Description : General Idea for grid generation and traversal while being O(1)
+         or as close as possible to it.
+         This just holds basic game logic
+ ============================================================================
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "grid.h"
+/*
+ * TODO : learn how turns work to create a 'move queue'
+ *
+ * TODO : REMOVE STDIO AND STDLIB WHEN FINISHED
+ *
+ * TODO : Create new branch + PR for character movement and turn queues
+ */
```

Comment:
Move these TODOs elsewhere, I suggest a TODO.txt thats formatted nicely for other files/TODOs